game_controller.py

```diff
@@ -237,6 +236,9 @@ def _clear_effects(self) -> None:
             ]
 
     def _resolve_effects(self) -> None:
+        for effect in self._game.effects:
```

Comment:
You've reworked resilve method but not activate method.
Activate should be changed too